src/cudadecoder/thread-pool-cia.h

```diff
@@ -0,0 +1,427 @@
+#pragma once
+
+#include <future>
+#include <memory>
+#include <thread>
+#include <type_traits>
+#include <queue>
+#include <vector>
+
+namespace kaldi {
+
+class join_threads {
```

Comment:
It seem like the class name not follow Kaldi naming convention

Reply:
Yes, it was adapted from a book that uses a different style: https://github.com/kaldi-asr/kaldi/pull/4811/files#diff-d472827499864b67e7925a3ea6f3b95d7b7cba4d0bc745a786c0bac6258fffc5R18-R19